### PR TITLE
Bugfix: Can update blocks in CSF instructions inline feedback

### PR DESCRIPTION
**Pull request by islemaster**
Bugfix needed for a playtest tomorrow.

When moving from one failure mode to another, the blocks in the feedback were not updating.

Before:
before

After:
after

The error message I was seeing was way down in the React renderer:
Screenshot from 2019-10-24 15-51-41

It seems like it might be [this bug](https://github.com/facebook/react/issues/7363) which should be fixed once we upgrade to React 16.

In the meantime, as a workaround I'm passing the message contents as a `key` to the feedback component, which forces it to unmount and remount when the message changes rather than trying to do a more targeted update.

I've filed a task to remove the workaround once React has been upgraded: https://codedotorg.atlassian.net/browse/LP-914

## Links

- [Slack conversation](https://codedotorg.slack.com/archives/CA3KCSGTD/p1571938769151100)
- [Jira](https://codedotorg.atlassian.net/browse/LP-912)

## Testing story

I've tested this manually on my machine and also added a unit test that would have caught this issue.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/templates/instructions/InstructionsCSFTest.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <InstructionsCSF {...DEFAULT_PROPS} />
+      </Provider>
+    );
+
+    failWithMessage('Repeat block: <xml><block type="controls_repeat"/></xml>');
+    assertFeedbackContainsText(wrapper, '>repeat</text>');
+
+    failWithMessage('If block: <xml><block type="controls_if"/></xml>');
+    assertFeedbackContainsText(wrapper, '>if</text>');
+  });
```
Comment (islemaster) [inline apps/test/unit/templates/instructions/InstructionsCSFTest.jsx]: Wow, it took a lot of painful setup to get this component rendering in a test.  Fortunately the actual test case is fairly compact.

File: apps/src/templates/instructions/InlineFeedback.jsx [hunk 2]
```diff
@@ -14,6 +14,14 @@ class InlineFeedback extends Component {
     styles: PropTypes.object
   };
 
+  /**
+   * Note: InlineFeedback will only convert XML to blockly when it is initially mounted, not on
+   * subsequent updates.  Recommended workaround while we are on React 15 is to add a changin `key`
```
Comment (Hamms) [inline apps/src/templates/instructions/InlineFeedback.jsx]:
```suggestion
   * subsequent updates.  Recommended workaround while we are on React 15 is to add a changing `key`
```

File: apps/test/unit/templates/instructions/InstructionsCSFTest.jsx (hunk 1 above)
Comment (Hamms) [reply apps/test/unit/templates/instructions/InstructionsCSFTest.jsx]: oof, yeah, this is a super weighty component

**Review by Hamms — APPROVED** [review]
Nice fix!

**Comment by islemaster** [comment]
Oh no!
```
InstructionsCSF | 1184s
-- | --
3445 | ✖ can change feedback when rendering different blockly blocks | 1184s
3446 | PhantomJS 2.1.1 (Linux 0.0.0) | 1184s
3447 | object tested must be an array, an object, or a string, but undefined given
```

This smells like a Phantom-specific issue, since it's not happening when I run tests in Chrome locally.  Can someone still able to run Phantom locally check this?

**Comment by islemaster** [comment]
@ajpal confirmed that this test is failing due to a Phantom-specific issue.  We need to get this fix out for tomorrow so I've added a commit removing the test, and we'll put it back tomorrow once we have a version that works on Phantom as well. (I suspect the complex CSS query I'm using.)

**Pull request merged**